pkgs/profpatsch/nman/nman.rs

```diff
@@ -137,28 +138,65 @@ fn build_man_page<'a>(drv: DrvWithOutput, section: &str, page: &str, tempdir: &T
 
     let mut path = PathBuf::from(OsStr::from_bytes(first_path));
     path.push("share/man");
-    path.push(format!("man{}", section));
-    path.push(page);
-
-    path.set_extension(format!("{}.gz", section));
-
-    if path.exists() {
-        Ok(Some(path))
-    } else {
-        // check for uncompressed man page if the derivation
-        // has no / a custom fixupPhase
-        path.set_extension(section);
-
-        if path.exists() {
-            Ok(Some(path))
-        } else {
-            Ok(None)
+
+    // no share/man, no man pages
+    if !path.exists() {
+        return Ok(None);
+    }
+
+    // expected sub directory of share/man or, if no section
+    // is given, all potential sub directories
+    let mut section_dirs: Vec<OsString> =
+        match section {
+            Some(s) => vec!(OsString::from(format!("man{}", s))),
+            None => {
+                std::fs::read_dir(path.as_path())
+                    .map_err(|_| NmanError::IO)?
+                    .filter_map(|entry| entry.ok())
+                    .map(|entry| entry.file_name())
+                    .collect()
+            },
+        };
+
+    // sorting should be ascending in terms of numerics,
+    // apart from that, not many requirements
+    section_dirs.sort();
+
+    const EXTENSIONS: [&str; 2] = [ ".gz", "" ];
+
+    for dir in section_dirs {
+        // separate "man" prefix from section indicator,
+        // while validating the particular sub directory
+        match dir.to_str().map(|d| d.split_at(3)) {
+            Some((_, "")) => continue,
+            Some(("man", s)) => {
+                // we have a valid man dir, check if it contains our page
+                path.push(dir.as_os_str());
+                path.push(page);
+
+                // for nix we almost always have .{section}.gz as extension,
+                // but traditionally plain .{section} is common and possible
+                for ext in EXTENSIONS.iter() {
+                    path.set_extension(format!("{}{}", s, ext));
+
+                    if path.exists() {
+                        return Ok(Some(path));
+                    }
+                }
+
+                // reset the PathBuf if we didn't find anything
+                path.pop(); // page
+                path.pop(); // section directory
```

Comment:
Eh, let’s clone for each part of the loop and leave this out.

Reply:
Done in 84ffc023c565e52f664f546dc12d63430d4be56a